haproxy-operator/templates/haproxy_route_grpc.cfg.j2

```diff
@@ -0,0 +1,23 @@
+{% set excluded_grpc_ports = [80, 443] %}
+{% for backend in backends if backend.external_grpc_port and backend.external_grpc_port not in excluded_grpc_ports and backend.servers | selectattr('protocol', 'eq', 'https') | list %}
```

Comment:
This should probably be a computed property in the charm code. I think we should try to move complex logic to the charm code and keep the templates simple.